### PR TITLE
bug fix: shutdown meta server before job manager init, may core

### PR DESCRIPTION
**Pull request by liuyu85cn**
shutdown job manager if its Initialization finished

### CODE REVIEW

File: src/daemons/MetaDaemon.cpp [hunk 1]
```diff
@@ -306,7 +306,9 @@ void signalHandler(int sig) {
             }
             {
                 auto gJobMgr = nebula::meta::JobManager::getInstance();
-                gJobMgr->shutDown();
+                if (gJobMgr) {
```
Comment (monadbobo) [inline src/daemons/MetaDaemon.cpp:309]: Here we also need to determine if gJobMgr has called the init function.

Comment (liuyu85cn) [reply src/daemons/MetaDaemon.cpp:309]: good catch

**Review by monadbobo — APPROVED** [review]
LGTM.

File: src/meta/test/JobManagerTest.cpp [hunk 2]
```diff
@@ -13,6 +13,8 @@
 #include "kvstore/Common.h"
 #include "webservice/WebService.h"
 
+#define private public
```
Comment (dangleptr) [inline src/meta/test/JobManagerTest.cpp:16]: What's this.....

Comment (liuyu85cn) [reply src/meta/test/JobManagerTest.cpp:18]: black magic...
this will make test code modify private member directly. 
without declare any "FRIEND_TEST"

**Review by dangleptr — DISMISSED** [review]

**Review by monadbobo — DISMISSED** [review]
LGTM.

**Review by bright-starry-sky — DISMISSED** [review]
LGTM

**Review by dangleptr — APPROVED** [review]

**Review by critical27 — APPROVED** [review]
Good catch

**Review by monadbobo — APPROVED** [review]
LGTM.

**Pull request merged**